components/common/src/command/package/install.rs

```diff
@@ -90,72 +199,60 @@ where
         product,
         version,
         fs_root_path.as_ref(),
-        cache_artifact_path.as_ref(),
+        artifact_cache_path.as_ref(),
         &cache_key_path,
-        ignore_target,
     )?;
 
-    if Path::new(ident_or_archive).is_file() {
-        task.from_artifact(ui, &Path::new(ident_or_archive))
-    } else {
-        task.from_ident(ui, PackageIdent::from_str(ident_or_archive)?, channel)
+    match *install_source {
+        InstallSource::Ident(ref ident) => task.from_ident(ui, ident.clone(), channel),
+        InstallSource::Archive(ref local_archive) => task.from_archive(ui, local_archive),
     }
 }
 
 struct InstallTask<'a> {
     depot_client: Client,
     fs_root_path: &'a Path,
-    cache_artifact_path: &'a Path,
+    /// The path to the local artifact cache (e.g., /hab/cache/artifacts)
+    artifact_cache_path: &'a Path,
     cache_key_path: &'a Path,
-    ignore_target: bool,
 }
 
 impl<'a> InstallTask<'a> {
-    pub fn new(
+    fn new(
         url: &str,
         product: &str,
         version: &str,
         fs_root_path: &'a Path,
-        cache_artifact_path: &'a Path,
+        artifact_cache_path: &'a Path,
         cache_key_path: &'a Path,
```

Comment:
Actually, I'll just rename to `key_cache_path`; switching back to `cache_artifact_path` got confusing because there is also a `cached_artifact_path()` function. We can just do a `core::fs` pass later.